src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -0,0 +1,50 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.domain.SessionImage.SessionImage;
+import nextstep.payments.domain.Payment;
+import nextstep.users.domain.NsUser;
+
+public class Session {
+    private Students students;
+    private Premium premium;
+    private SessionState sessionState;
+    private SessionImage sessionImage;
+
+    private int maxStudentCount;
+    private SessionDate sessionDate;
+
+
+
+
+    public Session(Premium premium, int maxStudentCount, SessionState sessionState, SessionDate sessionDate
+            ,SessionImage sessionImage) {
+        this.premium = premium;
+        this.students = new Students();
+        this.maxStudentCount = maxStudentCount;
+        this.sessionState = sessionState;
+        this.sessionDate = sessionDate;
+        this.sessionImage = sessionImage;
+
+    }
+
+    public void requestSession(Payment payment) {
+        validate();
+        validateSessionState();
+        premium.validateAmount(payment);
+        students.addStudent(payment.payingUser());
+    }
+
+    public void validate() {
+        if (premium.isPremium() && students.studentCount() >= maxStudentCount) {
+            throw new IllegalArgumentException("수강인원 초과");
+        }
+    }
+
+    private void validateSessionState() {
+        if (!SessionState.isRequestSession(this.sessionState)) {
```

Comment:
```suggestion
        if (!sessionState.isRequestSession()) {
```
State에 메시지를 던져서 확인하는 방향으로 만들면 어떨까요?

Reply:
enum 사용시 static말고 메세지를 던지는 방식은 생각을 안했었네요!
말씀하신대로 수정해보았습니다.